backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/projection/impl/FieldProjectionRequestContext.java

```diff
@@ -57,6 +59,17 @@ public void checkValidField(String absoluteFieldPath) {
 		}
 	}
 
+	@Override
+	public void checkNotNested(SearchQueryElementTypeKey<?> projectionKey, String hint) {
+		if ( absoluteCurrentFieldPath() != null ) {
+			throw log.cannotUseProjectionInNestedContext(
+					projectionKey.toString(),
+					hint,
+					EventContexts.indexSchemaRoot()
```

Comment:
As said in my earlier review:

> `indexSchemaRoot` definitely isn't correct since this exception happens only when we're not at the schema root. I'd remove the context (you don't have to provide one) or change it to EventContexts.fromIndexFieldAbsolutePath(...).

IMO `EventContexts.fromIndexFieldAbsolutePath(absoluteCurrentFieldPath())` would be fine here.